src/CQRS/LeanCode.CQRS.AspNetCore/Registration/CQRSObjectsRegistrationSource.cs

```diff
@@ -2,25 +2,23 @@
 using LeanCode.Components;
 using LeanCode.Contracts;
 using LeanCode.CQRS.Execution;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace LeanCode.CQRS.AspNetCore.Registration;
 
 internal class CQRSObjectsRegistrationSource
 {
-    private readonly TypesCatalog contractsCatalog;
-    private readonly TypesCatalog handlersCatalog;
+    private readonly IServiceCollection services;
+    private readonly HashSet<CQRSObjectMetadata> objects = new();
```

Comment:
I wonder if we should care more about equality comparison here - _technically_ you are now able to register one command with two handlers from separate assemblies.

This is a rather not probable case, but I'd compare only the object type here and silently ignore duplicates.